Reorganize Resources, top menu, and ohw23 & ohw22 menus

emiliom opened the pull request:
Substantial reorganization of pages and menus.

- Resources is now found in the top bar menu
- OHW22 was removed from the top bar menu
- Some content previously under the `resources` folder was moved and duplicated to both `ohw23` and `ohw22`: the `logistics` and `projects` folders
- Copied `tutorials` and `tutorials-index` folders from `ohw22` to `ohw23`. So, right now ohw23 tutorials is a stub, pointing to ohw22 tutorials
- Created stub ohw23 schedule page. Currently it's just a copy of the old ohw22 page

A bunch of event-specific content will need to be updated. Let's at least add warnings (admonitions) where they're found, while we do the updates.

@abkfenris : 
- The ohw22 menu is now broken. I tried different things and couldn't get it to work. Could you look into it?
- I haven't figured out how you link the `ohw-tutorials` repo into the `tutorials` folder. So, that part will be broken for `ohw23`

This PR is definitely not intended to be merged as is. Let's use it to discuss the reorganization and suggest changes large or small.

emiliom commented:
Once we've merged this PR, we can tackle PR #260

abkfenris commented:
For the OHW22 nav, I think that Sphinx only wants to render toctrees that are part of the global toctree hiearchy, so adding the OHW22 index explicitly to another tree should do it.

````md
# on about/pasthackweeks.md ?

```{toctree}
:maxdepth: 1

../ohw22/index.md
```
````

- https://sphinx-doc-zh.readthedocs.io/en/latest/markup/toctree.html
- https://sphinx-doc-zh.readthedocs.io/en/latest/config.html#confval-master_doc

The tutorials are linked via a git submodule which is defined [here](https://github.com/oceanhackweek/oceanhackweek.github.io/blob/source/.gitmodules), but there is an incantation for adding them that I don't remember at the moment, but [here is the PR](https://github.com/oceanhackweek/oceanhackweek.github.io/pull/199/files) when I set things up that way.

There are a whole bunch of other errors that I think correspond to changed pages that are probably worth going through.

emiliom commented:
Thanks @abkfenris ! I've pushed several commits that:
- Added ohw22/index.md to the `pasthackweeks.md` toctree. That fixed the ohw22 section menu issue!
- Did a bunch of other fixes to ensure toctrees are being generated as expected for ohw22 and ohw23
- Removed `logistics` and `projects` folders from `resources`, now that they've been moved to the event folders
- Moved `tutorials/getting_started.md` from `resources` to the individual event folders, renamed as `tutorials_getting_started.md`

The "tutorials-index" git submodule stuff for ohw23 is still pending. @abkfenris I'll let you take care of that.

How do things look now? FYI Old content is still largely not updated, with minor exceptions. I also for the most part haven't added any new warnings ("admonitions") about old content in individual pages. Once the page organization stabilizes and this PR is merged, maybe @alisonpchase and @btupper can tackle that?

abkfenris commented on index.md:
```diff
@@ -146,19 +146,14 @@ Thanks to our sponsors that have made OceanHackWeek possible over the last sever
-about/pasthackweeks.md
 OceanHackWeek 2023 <ohw23/index.md>
-OceanHackWeek 2022 <ohw22/index.md>
+resources/index.md
+about/pasthackweeks.md
 % posts
```
```suggestion
resources/index.md
about/pasthackweeks.md
```

Posts is throwing an error due to the mix of rst and md formatting.

emiliom replied:
Ah. I'd seen that error but didn't try to figure it out.

emiliom replied:
I've fixed this.

abkfenris commented on ohw22/logistics/getting_help.md:
It feels like most of getting help and the projects info right now are more general and could live in resources.

emiliom replied:
The mix of general and event-specific content seemed more even in getting help and projects info. It's fair to say that both follow a template where some specific information (Slack channel names, dates, brief descriptions, etc) would be customized in each event. For the getting help page, the text is the same except for a bunch of Slack channel names that are spread throughout the page. I think that makes the use of admonitions too crowded / bloated.

Anyways, I'll take another look.

abkfenris commented on ohw23/logistics/getting_help.md:
If we have most of the getting help info in resources, we can include parts or all of another file with `include` or `literalinclude` blocks for duplicated info.

I use some here: https://github.com/xpublish-community/xpublish/blob/418c660398a1b48ae73f9e2c04158ad1c83b6de6/docs/source/getting-started/tutorial/dataset-router-plugin.md?plain=1#L7-L12

emiliom replied:
That will be helpful, thanks!

emiliom replied:
`I'm using an `include` block now, in the event projects page. Thanks again.

abkfenris reviewed (commented):
There are a few pages that have been moved from resources to event specific pages and duplicated. I think we can keep them in resources and use admonitions (and maybe tabs) to call out any event specifics.

emiliom commented:
- I've moved the general project content to `resources/projects.md`, and the event-specific project content to `ohw23/projects/index.md` and `ohw22/projects/index.md`. Those two event pages now use an `include` block to load `resources/projects.md`
- Let's leave the logistics pages in the event directories for now, duplication and all

There are still some errors that could be fixed w/o too much trouble. Maybe later!

emiliom commented:
- Added prominent admonition in OHW23 front page about some ohw23 content being currently old
- Purged links to ohw22 tutorials, and left in place only the getting started page for now, until ohw23 tutorials land
